src/main/scala/analysis/InterprocSteensgaardAnalysis.scala

```diff
@@ -0,0 +1,389 @@
+
+  private def nextMallocCount() = {
+    mallocCount += 1
+    s"malloc_$mallocCount"
+  }
+
+  /**
+   * In expressions that have accesses within a region, we need to relocate
+   * the base address to the actual address using the relocation table.
+   * @param address
+   * @return BitVecLiteral: the relocated address
+   */
+  def relocatedBase(address: BitVecLiteral): BitVecLiteral = {
+    val tableAddress = globalOffsets.getOrElse(address.value, address.value)
+    BitVecLiteral(tableAddress, address.size)
+  }
+
+  /**
+   * Used to reduce an expression that may be a sub-region of a memory region.
+   * Pointer reduction example:
+   * R2 = R31 + 20
+   * Mem[R2 + 8] <- R1
+   *
+   * Steps:
+   * 1) R2 = R31 + 20         <- ie. stack access (assume R31 = stackPointer)
+   *         ↓
+   *    R2 = StackRegion("stack_1", 20)
+   *
+   * 2) Mem[R2 + 8] <- R1     <- ie. memStore
+   *         ↓
+   *    (StackRegion("stack_1", 20) + 8) <- R1
+   *         ↓
+   *    MMM.get(20 + 8) <- R1
+   *
+   * @param binExpr
+   * @param n
+   * @return Set[MemoryRegion]: a set of regions that the expression may be pointing to
+   */
+  def reducibleToRegion(binExpr: BinaryExpr, n: CfgCommandNode): Set[MemoryRegion] = {
+    var reducedRegions = Set.empty[MemoryRegion]
+    binExpr.arg1 match {
+      case variable: Variable =>
+        val reg = RegisterVariableWrapper(variable)
+        val ctx = regionAccesses(n)
+        if (ctx.contains(reg)) {
+          ctx(reg) match {
+            case FlatEl(al) =>
+              val regions = al match {
+                case loadL: MemoryLoad =>
+                  exprToRegion(loadL.index, n)
+                case _ =>
+                  exprToRegion(al, n)
+              }
+              val results = evaluateExpressionWithSSA(binExpr.arg2, constantProp(n))
+              for {
+                b <- results
+                r <- regions
+              } {
+                r match {
+                  case stackRegion: StackRegion =>
+                    val nextOffset = BinaryExpr(BVADD, stackRegion.start, b)
+                    evaluateExpressionWithSSA(nextOffset, constantProp(n)).foreach { b2 =>
+                      reducedRegions ++= exprToRegion(BinaryExpr(BVADD, stackPointer, b2), n)
+                    }
+                  case dataRegion: DataRegion =>
+                    val nextOffset = BinaryExpr(BVADD, relocatedBase(dataRegion.start), b)
+                    evaluateExpressionWithSSA(nextOffset, constantProp(n)).foreach { b2 =>
+                      reducedRegions ++= exprToRegion(b2, n)
+                    }
+                  case _ =>
+                }
+              }
+          }
+        }
+        evaluateExpressionWithSSA(binExpr, constantProp(n)).foreach { b =>
+          val region = mmm.findDataObject(b.value)
+          reducedRegions = reducedRegions ++ region
+        }
+      case _ =>
+    }
+    reducedRegions
+  }
```

Comment:
This seems to just assume that the BinaryExpr is an addition operation?

Reply:
It could be replaced with `binExpr.op` I can change that